discoverd/deployment/deployment.go

```diff
@@ -0,0 +1,196 @@
+package deployment
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/attempt"
+	"github.com/flynn/flynn/pkg/stream"
+)
+
+type DeploymentState string
+
+const (
+	DeploymentStatePerforming DeploymentState = "performing"
+	DeploymentStateDone       DeploymentState = "done"
+)
+
+func NewDeployment(service string) (*Deployment, error) {
+	s := discoverd.NewService(service)
+	events := make(chan *discoverd.Event)
+	stream, err := s.Watch(events)
+	if err != nil {
+		return nil, err
+	}
+	return &Deployment{s, events, stream}, nil
+}
+
+// Deployment is a wrapper around service metadata for marking jobs as either
+// performing a deployment or done deploying
+type Deployment struct {
+	service discoverd.Service
+	events  chan *discoverd.Event
+	stream  stream.Stream
+}
+
+const deploymentTimeout = time.Minute
```

Comment:
Actually I am not sure about this. The timeout should probably be configurable but it's not clear of `app.DeployTimeout` is directly comparable to the `deploymentTimeout` used here as it's used to both determine the amount of time we spend trying to mark ourselves performing and how long we wait for all `done` events to arrive. I am thinking if we do want to use app.DeploymentTimeout it should be given as an argument to `Wait()` as it's the only place this information is needed or even relevant.
The same could probably be said for the mark performing stage, it should be up to the application to choose how long is reasonable enough to wait for other nodes to finish.